static_frame/core/frame.py

```diff
@@ -6657,68 +6657,78 @@ def unset_index(self, *,
             consolidate_blocks:
             columns_constructors:
         '''
+        # disallows specifying names with 'drop=True'
+        if drop and names:
```

Comment:
Subtle but by doing this `is` check we implicitly check that the drop argument is a Boolean; that way another truthy value (like `'keep'`!) cannot be used.

```suggestion
        if drop is True and names:
```

Reply:
Ah nice.